python/tank/bootstrap/cached_configuration.py

```diff
@@ -276,6 +281,24 @@ def update_configuration(self):
                     self.path.as_shotgun_dict()
                 )
 
+    def _delete_old_backups(self, latest_backup_folder_path):
+        """
+        Deletes all of the folders in the backups_folder except
+        for the last backup created.
+        """
+    
+        if os.path.exists(latest_backup_folder_path):
```

Comment:
please add some debug logging so that the log file tells the story of what is being deleted!